app/models/piece.rb

```diff
@@ -1,53 +1,93 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
 
-  # Returns true/false if a piece is obstructed in its movement.
-  # Errors for movements not diagonal, vertical or horizontal
+  def move_to!(x, y)
+  # Immediately return to controller with false if move is deemed invalid
+    return false if !valid_move? x, y
+
+  # Return to controller with false if moving piece is obstructed
+    return false if is_obstructed? x, y
```

Comment:
I don't understand what you're emphasizing. I'm aware that all pieces (though technically not the Knight) can be obstructed.